lib/keccak/keccak.c

```diff
@@ -286,7 +286,13 @@ __attribute__((target("bmi,bmi2"))) static void keccakf1600_bmi(uint64_t state[2
 
 __attribute__((constructor)) static void select_keccakf1600_implementation()
 {
-    if (__builtin_cpu_supports("bmi2"))
+    // Init CPU information.
+    // This is needed on macOS because of the bug: https://bugs.llvm.org/show_bug.cgi?id=48459.
+    __builtin_cpu_init();
+
+    // Check if both BMI and BMI2 are supported. Some CPUs like Intel E5-2697 v2 incorrectly
```

Comment:
Do you have some kind of bug report / errata link for this?

Reply:
No. This comes from experiments on CI.

Reply:
I can't even find any details spec of E5-2697 v2. I checks the assembly code generated by `__builtin_cpu_supports("bmi2")` and it looks correct. I also found a case where for some other Intel CPU it was reporting some features not actually available. That was fixed by BIOS update :)
Seems like a CPU bug, but no more information than this.

Reply:
> I also found a case where for some other Intel CPU it was reporting some features not actually available. That was fixed by BIOS update :)

That sounds like a new microcode was pushed to the BIOS. The microcode must be loaded on every cold start.